software/python/state.py

```diff
@@ -0,0 +1,153 @@
+from enum import Enum
+import cv2
+import time
+from constants import *
+import numpy as np
+
+
+class State(Enum):
+    OPP_BASKET = 0
+    BALL = 1
+    THROW_GOAL = 2
+
+
+class StateHandler:
+    def __init__(self, cap, robot, opp_thresholder, pole_thresholder, ball_thresholder, ball_detector, pole_detector, map_to_max, orbit_left, orbit_right, delay_camera, go_forward):
+        self.cap = cap
+        self.robot = robot
+        self.opp_thresholder = opp_thresholder
+        self.pole_thresholder = pole_thresholder
+        self.ball_thresholder = ball_thresholder
+        self.ball_detector = ball_detector
+        self.pole_detector = pole_detector
+        self.set_state(State.BALL)
+        self.state_start_time = time.time()
+        self.map_to_max = map_to_max
+        self.orbit_left = orbit_left
+        self.orbit_right = orbit_right
+        self.delay_camera = delay_camera
+        self.go_forward = go_forward
+    
+
+    def img_processing(self):
+        frame, depth_frame = self.cap.get_frames()
+        self.frame_hsv = cv2.cvtColor(frame, cv2.COLOR_BGR2HSV)
+        self.depth_frame = depth_frame
+    
+
+    def state_handler(self):
+        self.img_processing()
+        if (self.get_state() == State.BALL):
+            self.state_ball()
+        elif (self.get_state() == State.OPP_BASKET):
+            self.state_go_to_opponent_basket()
+        elif (self.get_state() == State.THROW_GOAL):
+            self.state_throw_to_goal()
+        else:
+            raise Exception("Unexpected state")
+        
+        cv2.imshow("Aken", self.frame_hsv)
+
+
+    def get_state(self):
+        return self.state
+
+    
+    def set_state(self, state):
+        self.state = state
+        self.state_start_time = time.time()
+
+
+    def state_go_to_opponent_basket(self):
+        if time.time() - self.state_start_time > STATE_OVERFLOW_TIME:
```

Comment:
Is this the only way to exit this state? What when you see a ball? Dot you want to chase it?

Reply:
Purpose of this task is to change the location and find new balls in this location, if there aren't any currently. It's good if we see ball at some point, but we will see some balls after moving as well so it doesn't matter so much